### ISSUE TITLE
Clang should warn about `noexcept(noexcept(is_meow_v))` always being true

### ISSUE THREAD

**Issue by StephanTLavavej** [llvm/llvm-project#61253]
Consider the following code (compiled with Clang 15.0.1). Clang is behaving correctly as required by the Standard here, but there’s a notorious bug in the user’s code that the compiler could warn about:

```
C:\Temp>type meow.cpp
```
```cpp
#include <iostream>
#include <string>
#include <type_traits>
using namespace std;

template <typename T>
void do_thing_good(const T& t) noexcept(is_nothrow_copy_constructible_v<T>) {
    [[maybe_unused]] T duplicate = t;
}

// USER BUG: noexcept(noexcept(is_nothrow_copy_constructible_v<T>)) is always true!
template <typename T>
void do_thing_bad(const T& t) noexcept(noexcept(is_nothrow_copy_constructible_v<T>)) {
    [[maybe_unused]] T duplicate = t;
}

int main() {
    const string cat{"cat"};

    cout << boolalpha;

    cout << R"(noexcept(do_thing_good(123)): )" << noexcept(do_thing_good(123)) << "\n";
    cout << R"(noexcept(do_thing_good(cat)): )" << noexcept(do_thing_good(cat)) << "\n";

    cout << R"(noexcept(do_thing_bad(123)): )" << noexcept(do_thing_bad(123)) << "\n";
    cout << R"(noexcept(do_thing_bad(cat)): )" << noexcept(do_thing_bad(cat)) << "\n";
}
```
```
C:\Temp>clang-cl /EHsc /nologo /W4 /std:c++latest meow.cpp && meow
noexcept(do_thing_good(123)): true
noexcept(do_thing_good(cat)): false
noexcept(do_thing_bad(123)): true
noexcept(do_thing_bad(cat)): true
```

The bug is accidentally saying `noexcept(noexcept(variable template type trait))`, when only one `noexcept` should be present. This mistake is *extremely* easy to make (as `noexcept(noexcept(arbitrary expression))` is commonly needed elsewhere).

The compiler could warn about this, for example whenever it sees `noexcept(noexcept(THING))` being applied to something that’s a `constexpr bool`. False positives seem quite unlikely, and the value of the warning would outweigh any potential need for rare suppressions.

**Comment by TheInferentialImp** [comment]
Could be done as:

`warning: expression in noexcept specification is a constant expression; redundant noexcept clause [-Wredundant-noexcept]
void do_thing_bad(const T& t) noexcept(noexcept(is_nothrow_copy_constructible_v<T>)) {
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
`

or in arithmetic form:

`char *reverse(char *str) {
    size_t len = strlen(str);
    char *start = str;
    char *end = str + len - 1;
    while (start < end) {
        char temp = *start;
        *start = *end;
        *end = temp;
        ++start;
        --end;
    }
    return str;
}
` 
sorry about my indentations, don't seem to be working

**Comment by frederick-vs-ja** [comment]
It seems reasonable to warn on all `noexcept(noexcept(id-expression))` for `noexcept` specifier, and all `noexcept(noexcept(expression))` for `noexcept` operator (such an occurence is likely a copy-pasta).

Maybe related: [A modest proposal for a GCC diagnostic](https://quuxplusone.github.io/blog/2018/04/09/long-long-long-is-too-long-for-gcc/).
